src/language/nodes.py

```diff
@@ -272,6 +272,27 @@ def get_add_methods(self):
                             auto last_it = const_iter_cast({self.varname}, last);
                             return {self.varname}.erase(first_it, last_it);
                          }}
+                         /**
+                          * \\brief Erase non-consecutive members to {self.varname}
+                          */
+                         void erase_{to_snake_case(self.class_name)}(const std::unordered_set<{self.class_name}*>& to_be_erased) {{
+                            // loosely following the cpp reference of remove_if
+
+                            auto first = {self.varname}.begin();
+                            auto last = {self.varname}.end();
+                            auto result = first;
+                        
+                            while (first != last) {{
+                                if (to_be_erased.find(first->get()) == to_be_erased.end()) {{
```

Comment:
`<nitpick>`Unless I'm mistaken what you do here is move all the elements out of the way that should be kept, and then you delete the rest. Wouldn't be more efficient to do the opposite? Only move elements that should be deleted?

Reply:
Example: I need to remove an element at the beginning of the vector. If I call std::erase(begin()), the code moves all the other elements at the beginning and then erases the last one. In fact [here](http://www.cplusplus.com/reference/vector/vector/erase/) is stated that the complexity is:

> Linear on the number of elements erased (destructions) plus the number of elements after the last element deleted (moving).

So, in any case I need to move the remaining elements at the beginning and then erase the back part. The algorithm that I use is very similar to the one stated for [remove_if](http://www.cplusplus.com/reference/algorithm/remove_if/) as explained in the comment